### PR TITLE
document usage of openMP in the manual

### LINKED ISSUE #4071: Is there any docs I can find to understand when parallelism is used
Hello, this is a question as opposed to an issue, please kindly refer me to the proper channel if that was not the right one.
I am looking for documentation to understand exactly where data.table uses openMP parallelism, there are cases where I clearly see the parallelism kick in but in cases where I do long `by` operations it is clearly not.
Is there anything I can use ?

Regards

### PR DESCRIPTION
**Pull request by MichaelChirico**
Closes #4071

### CODE REVIEW

File: man/openmp-utils.Rd [hunk 1]
```diff
@@ -30,6 +32,23 @@
   \code{setDTthreads()} affects \code{data.table} only and does not change R itself or other packages using OpenMP. We have followed the advice of section 1.2.1.1 in the R-exts manual: "\ldots or, better, for the regions in your code as part of their specification\ldots num_threads(nthreads)\ldots That way you only control your own code and not that of other OpenMP users." Every parallel region in data.table contain a \code{num_threads(getDTthreads())} directive. This is mandated by a \code{grep} in data.table's quality control script.
 
   \code{setDTthreads(0)} is the same as \code{setDTthreads(percent=100)}; i.e. use all logical CPUs, subject to \code{Sys.getenv("OMP_THREAD_LIMIT")}. Please note again that CRAN's daily test system sets \code{OMP_THREAD_LIMIT} to 2, so developers of CRAN packages should never change \code{OMP_THREAD_LIMIT} inside their package to a value greater than 2.
+
+  Internally parallelized code is used in the following places:
+
+  \itemize{
+    \item{\code{between.c} - \code{\link{between}()}}
```
Comment (jangorecki) [inline man/openmp-utils.Rd]: Extra `()` everywhere is redundant

Comment (MichaelChirico) [reply man/openmp-utils.Rd]: maybe. I wanted to emphasize it's a function. also noticed this style in base. feel free to remove

File: .dev/CRAN_Release.cmd [hunk 2]
```diff
@@ -39,6 +39,9 @@ grep --exclude="./src/openmp-utils.c" omp_get_max_threads ./src/*
 # Ensure all #pragama omp parallel directives include a num_threads() clause
 grep "pragma omp parallel" ./src/*.c | grep -v getDTthreads
 
+# Update documented list of places where openMP parallelism is used: c.f. ?openmp
+grep -Elr "#pragma omp (?:for|parallel)" src | sort
```
Comment (jangorecki) [inline .dev/CRAN_Release.cmd]: @MichaelChirico we also use `_Pragma("omp parallel for num_threads(getDTthreads())")` so probably better to remove leading `#`

Comment (mattdowle) [reply .dev/CRAN_Release.cmd]: Nice spot. Changed here: 5730cf8f96c3d9febb4b144976965f29382741ae. It caught one more file (freadR.c, as well as fread.c, making 17 files total) so nothing else to update.

**Pull request merged**